src/Type/Regex/RegexAstWalkResult.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Regex;
+
+/** @immutable */
+final class RegexAstWalkResult
+{
+
+	/**
+	 * @param array<int, RegexCapturingGroup> $capturingGroups
+	 * @param list<string> $markVerbs
+	 */
+	public function __construct(
+		private int $alternationId,
+		private int $captureGroupId,
+		private array $capturingGroups,
+		private array $markVerbs,
+	)
+	{
+	}
+
+	public static function createEmpty(): self
+	{
+		return new self(
+			-1,
+			100,
```

Comment:
I know you moved the code only, but why 100?

Reply:
in the initial implementation there we so many arrays involved, that I used a different start-index to trigger php warnings in case I mixed up the keys between different arrays ;)

Reply:
please add a comment into the code